bin/install_swiftpm_dependencies

```diff
@@ -1,38 +1,89 @@
 #!/bin/bash -eu
 
 print_usage() {
-  echo "Usage:"
-  echo "  Using the \`Package.resolved\` of an Xcode \`.xworkspace\`:"
-  echo "    ${0##*/} --workspace PATH"
-  echo "  Using the \`Package.resolved\` of an Xcode \`.xcodeproj\`:"
-  echo "    ${0##*/} --project PATH"
-  echo "  Using the \`Package.resolved\` at the root, managed by \`swift package\` instead of Xcode:"
-  echo "    ${0##*/} --use-spm"
+  cat <<EOF
+Usage:
+  Install SwiftPM dependencies using the \`Package.resolved\` file:
+
+  1. For an Xcode \`.xcworkspace\`:
+       ${0##*/} --workspace PATH [--strict-package-check]
+
+  2. For an Xcode \`.xcodeproj\`:
+       ${0##*/} --project PATH [--strict-package-check]
+
+  3. For a Swift package managed by \`swift package\` instead of Xcode:
+       ${0##*/} --use-spm [--strict-package-check]
+
+  4. Default behavior (Attept to find an appropriate workspace, project, or swift package):
+       ${0##*/} [--strict-package-check]
+
+Options:
+  --workspace PATH          Specify the path to the Xcode workspace.
+  --project PATH            Specify the path to the Xcode project.
+  --use-spm                 Use the Swift package manager at the root.
+  --strict-package-check    Fail the installation if the \`Package.resolved\` file has uncommitted changes or is out of date.
+                            If specified more than once, the last value will be used.
+
+Examples:
+  Install dependencies for an Xcode workspace:
+      ${0##*/} --workspace MyApp.xcworkspace
+
+  Install dependencies with strict package check:
+      ${0##*/} --project MyApp.xcodeproj --strict-package-check
+
+  Install dependencies for a Swift package at the root:
+      ${0##*/} --use-spm
+EOF
 }
 
 # Parse input parameters
 XCWORKSPACE_PATH=
 XCODEPROJ_PATH=
 USE_SPM=false
-
-case ${1:-} in
-  --workspace)
-    XCWORKSPACE_PATH="${2:?you need to provide a value after the --workspace option}"
-    shift; shift
-    ;;
-  --project)
-    XCODEPROJ_PATH=${2:?you need to provide a value after the --project option}
-    shift; shift
-    ;;
-  --use-spm)
-    USE_SPM=true
-    shift
-    ;;
-  -h|--help)
-    print_usage
-    exit 0
-    ;;
-esac
+STRICT_PACKAGE_CHECK=false
+
+# Track the number of targets found in the arguments
+TARGET_COUNT=0
```

Comment:
Seems this variable is now completely unused (I'm guessing it was initially introduced as an attempt to detect argument conflicts but then you switched to test for `if [[ -n $XCWORKSPACE_PATH || -n $XCODEPROJ_PATH || $USE_SPM == "true" ]]` in each branch since?